MekHQ/src/mekhq/campaign/handler/PostScenarioDialogHandler.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2024 - The MegaMek Team. All Rights Reserved.
+ *
+ *  This file is part of MekHQ.
+ *
+ *  MekHQ is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  MekHQ is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mekhq.campaign.handler;
+
+import mekhq.MekHQ;
+import mekhq.campaign.Campaign;
+import mekhq.campaign.Kill;
+import mekhq.campaign.ResolveScenarioTracker;
+import mekhq.campaign.event.ScenarioResolvedEvent;
+import mekhq.campaign.mission.AtBScenario;
+import mekhq.campaign.personnel.Person;
+import mekhq.campaign.personnel.autoAwards.AutoAwardsController;
+import mekhq.campaign.personnel.enums.PersonnelStatus;
+import mekhq.gui.CampaignGUI;
+import mekhq.gui.dialog.RetirementDefectionDialog;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * @author Luana Coppio
+ */
+public class PostScenarioDialogHandler {
```

Comment:
I'm a little concerned about this class. With this implementation we have, effectively, three separate ways to conclude a scenario with slightly different implementations: concluding a mm scenario and going back to mhq; resolving manually via the Briefing Tab; and now this.

My concern is that without any centralized system if someone wants to add a new action to occur during scenario resolution they now need to update three different locations, without anything in the code inherently pointing them to do so. It's already bad enough that we have two different resolution systems.

I'm not sure what we want to do to resolve this, so this is more just to highlight a potential issue than requesting specific change.

Reply:
Oh, I knew I was missing something, this class also substitute `BriefingTab#resolveScenario`. let me look back into it.